File: src/Stronghold.Application/Features/Orders/OrderResponse.cs
namespace Stronghold.Application.Features.Orders;

public class OrderResponse
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string UserName { get; set; } = string.Empty;
    public decimal TotalAmount { get; set; }
    public string DeliveryAddress { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public string? StripePaymentIntentId { get; set; }
    public string? ClientSecret { get; set; }
    public DateTime CreatedAt { get; set; }
    public List<OrderItemResponse> Items { get; set; } = new();
}

File: stronghold_desktop/lib/features/orders/models/order_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'order_response.freezed.dart';
part 'order_response.g.dart';

@freezed
abstract class OrderResponse with _$OrderResponse {
  const factory OrderResponse({
    required int id,
    required int userId,
    required String userName,
    required double totalAmount,
    required String deliveryAddress,
    required String status,
    String? stripePaymentIntentId,
    String? clientSecret,
    required DateTime createdAt,
    @Default([]) List<OrderItemResponse> items,
  }) = _OrderResponse;

  factory OrderResponse.fromJson(Map<String, dynamic> json) =>
      _$OrderResponseFromJson(json);
}

@freezed
abstract class OrderItemResponse with _$OrderItemResponse {
  const factory OrderItemResponse({
    required int id,
    required int productId,
    required String productName,
    String? productImageUrl,
    required int quantity,
    required double unitPrice,
    required double subtotal,
  }) = _OrderItemResponse;

  factory OrderItemResponse.fromJson(Map<String, dynamic> json) =>
      _$OrderItemResponseFromJson(json);
}

@freezed
abstract class PagedOrderResponse with _$PagedOrderResponse {
  const factory PagedOrderResponse({
    required List<OrderResponse> items,
    required int totalCount,
    required int totalPages,
    required int currentPage,
    required int pageSize,
  }) = _PagedOrderResponse;

  factory PagedOrderResponse.fromJson(Map<String, dynamic> json) =>
      _$PagedOrderResponseFromJson(json);
}
